op-challenger/game/fault/trace/prestates/multi.go

```diff
@@ -75,8 +76,13 @@ func (m *MultiPrestateProvider) fetchPrestate(hash common.Hash, fileType string,
 	if resp.StatusCode != http.StatusOK {
 		return fmt.Errorf("%w from url %v: status %v", ErrPrestateUnavailable, prestateUrl, resp.StatusCode)
 	}
-	tmpFile := dest + ".tmp" + fileType // Preserve the file type extension so compression is applied correctly
-	out, err := ioutil.NewAtomicWriterCompressed(tmpFile, 0o644)
+	tmpFile := dest + ".tmp" + fileType // Preserve the file type extension so state decoding is applied correctly
+	var out *ioutil.AtomicWriter
+	if strings.HasSuffix(fileType, ".gz") {
+		out, err = ioutil.NewAtomicWriter(tmpFile, 0o644)
+	} else {
+		out, err = ioutil.NewAtomicWriterCompressed(tmpFile, 0o644)
+	}
```

Comment:
Shouldn't `NewAtomicWriterCompressed` handle this automatically based on the file type? (the name is potentially a bit misleading - it's really NewAtomicWriterMaybeCompressedDependendingOnFileType)

Reply:
The MT-Cannon state is already compressed (mt-cannon.state.gz). But given the file extension, the `NewAtomicWriterCompressed` file type detection assumes you want compression enabled so it re-compresses the state.  
This breaks the MT state decoder.
We still need to preserve the file extension so that the MT state decoder knows how to read the file.

The alternative is require MT prestates to be in their raw, uncompressed, format. This is a bit awkward as the current tooling emits compressed prestates.

Reply:
oh interesting... if the content is always appropriately compressed or not then we should just use `NewAtomicWriter` all the time.

Reply:
Only MT-Cannon uses a compressed prestate. The logic is awkward either way (and breaking a couple assumptions in various tests) so it's probably easier to simply assume the prestates are uncompressed.

Reply:
We support .json.gz downloads as well for single threaded. Given the source file name and name we download to always match, it we shouldn't need to change the compression of what we download (hence just use `NewAtomicWriter`).  The tests are quite likely wrong about that though given I hadn't considered the fact that the source data was already compressed when writing this, so likely tested that it does exactly the wrong thing too.

Reply:
Yeah confirmed the tests are broken - they're always returning uncompressed content even when the request was for a `.gz` file which would be compressed on the server.